indexprovider/wrapper.go

```diff
@@ -77,29 +85,180 @@ func NewWrapper(cfg *config.Boost) func(lc fx.Lifecycle, h host.Host, r repo.Loc
 			enabled:        !isDisabled,
 			piecedirectory: piecedirectory,
 			bitswapEnabled: bitswapEnabled,
+			ssm:            ssm,
 		}
-		w.usm = NewUnsealedStateManager(w, legacyProv, dealsDB, ssDB, storageService, w.cfg.Storage)
 		return w, nil
 	}
 }
 
 func (w *Wrapper) Start(ctx context.Context) {
 	w.prov.RegisterMultihashLister(w.MultihashLister)
 
-	runCtx, runCancel := context.WithCancel(context.Background())
+	runCtx, runCancel := context.WithCancel(ctx)
 	w.stop = runCancel
 
-	// Watch for changes in sector unseal state and update the
-	// indexer when there are changes
-	go w.usm.Run(runCtx)
-
 	// Announce all deals on startup in case of a config change
 	go func() {
 		err := w.AnnounceExtendedProviders(runCtx)
 		if err != nil {
 			log.Warnf("announcing extended providers: %w", err)
 		}
 	}()
+
```

Comment:
The changeset below is not modified and belonged to the `Unsealed State Manager` but it is really specific to the Index Provider. So I split it away from the Sector State Manager, and kept it under the Index Provider.